docs/docs/architecture/security.md

```diff
@@ -0,0 +1,105 @@
+For authorization, the [manifest](../features/manifest.md) defines the Marble's access to secrets and keys after successful attestation.
+Furthermore, MarbleRun's' [RBAC](../workflows/define-manifest.md#roles) attaches [users'](../workflows/define-manifest.md#users) identities to roles in the manifest.
+Each role is associated with a set of operations the user can perform in the deployment.
+Users are authenticated by the Coordinator using an RSA or ECDSA public key defined in the manifest.
+
+## Public Key Infrastructure and Certificate Authority
+
+The Coordinator establishes a public key infrastructure (PKI) for MarbleRun and acts as the Certificate Authority (CA).
+The goal of the PKI is to make authentication of confidential applications based on remote attestation accessible and usable.
+The Coordinator provides an [API](../reference/coordinator.md) for retrieving an SGX attestation statement that embeds its *Root CA Certificate* in the user-defined body.
+By verifying the statement, clients can verify the certificate's authenticity and, thereby, the MarbleRun CA.
+See the [attested TLS](#attested-tls-atls) section for details behind that concept.
+All MarbleRun clients and Marbles can then use the attested *Root CA Certificate* for authenticating TLS connections.
+This is further illustrated conceptionally in the [attestation](../features/attestation.md) section. The following focuses on the cryptographic primitives.
+The Coordinator generates a root x509 certificate and corresponding asymmetric key pair during initialization.
+The [Elliptic Curve Digital Signature Algorithm (ECDSA)](https://www.secg.org/sec1-v2.pdf#page=49) is used with curve [P256](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf#page=111).
+The *Root CA Certificate* has no expiry date and lives as long as the MarbleRun deployment.
+
+Alongside the *Root CA Certificate*, the Coordinator generates an x509 *Intermediate Certificate* and corresponding asymmetric key pair, again using ECDSA with P256.
+The *Intermediate Certificate* is signed by the Coordinator's *Root CA Certificate* and rotated with every manifest update.
+When you push an update to the manifest (for example, bump up the *SecurityVersion* of a Mable), the *Intermediate Certificate* will change.
+Instances with the new version won't authenticate with instances of the old version and vice versa.
+Hence, no data flow is happening between different *SecurityVersions* of your application.
+However, the *Root CA Certificate* doesn't change. So you can still verify the Coordinator and your application from the outside and ensure it's the same instance you might have interacted with.
+Applications interacting with the MarbleRun deployment can decide whether to use the intermediate or the root certificate as CA, depending on whether they want to notice and explicitly handle manifest updates.
```

Comment:
I'm not sure if the average user gets the implications of this approach. Should we describe it more explicit? Like

```suggestion
Applications interacting with the MarbleRun deployment should use the intermediate certificate as CA to make manifest updates observable:
if the manifest changes, connections to the deployment will fail.
The application owner can then review the changes and install the new intermediate certificate to approve them.
If such observability isn't required, the application can use the root certificate as CA.
In that case, the application will continue to work even if the manifest changes.
```